compiler/rustc_middle/src/ty/print/mod.rs

```diff
@@ -129,7 +129,12 @@ pub trait Printer<'tcx>: Sized {
             }
 
             _ => {
-                let parent_def_id = DefId { index: key.parent.unwrap(), ..def_id };
+                let parent_def_id = match self.tcx().opt_rpitit_info(def_id) {
+                    Some(ImplTraitInTraitData::Trait { fn_def_id, .. })
+                    | Some(ImplTraitInTraitData::Impl { fn_def_id }) => fn_def_id,
+
+                    _ => DefId { index: key.parent.unwrap(), ..def_id },
+                };
```

Comment:
In this case:

```
trait Foo {
  fn a() -> impl Sized;
  fn b() -> impl Sized;
```

The second associated type has the assoc-item name `b::{opaque#0}` (from the `impl Sized` it comes from), but is disambiguated in the trait as `{opaque#1}` (since it's the second synthesized RPITIT that is lowered), so it'll be pretty printed like `b::{opaque#1}` here, I think.

Reply:
Oh, yeah that's correct. Should we fix that or leave it as `Foo::{opaque#0}` and so on?.

Reply:
Maybe we would prefer to do this https://github.com/rust-lang/rust/pull/109499/commits/9dafbde6b86c8fbd2f0542161fbbbe3a3d04e2ed ?